mobile/Screen/MapScreen.js

```diff
@@ -1,39 +1,265 @@
+        index = state.markers.length - 1;
+      }
+      if (index <= 0) {
+        index = 0;
+      }
+
+      clearTimeout(mapRef.regionTimeout);
+      mapRef.regionTimeout = setTimeout(() => {
+        if (mapRef.index !== index) {
+          mapRef.index = index;
+          const { coordinate } = state.markers[index];
+          mapRef.current.animateToRegion(
+            {
+              ...coordinate,
+              latitudeDelta: state.region.latitudeDelta,
+              longitudeDelta: state.region.longitudeDelta,
+            },
+            350,
+          );
+        }
+      }, 10);
+    });
+  }, []);
+
+  const interpolations = state.markers.map((marker, index) => {
+    const inputRange = [
+      (index - 1) * CARD_WIDTH,
+      index * CARD_WIDTH,
+      (index + 1) * CARD_WIDTH,
+    ];
+    const scale = mapRef.animation.interpolate({
+      inputRange,
+      outputRange: [1, 2.5, 1],
+      extrapolate: 'clamp',
+    });
+    const opacity = mapRef.animation.interpolate({
+      inputRange,
+      outputRange: [0.35, 1, 0.35],
+      extrapolate: 'clamp',
+    });
+    return { scale, opacity };
+  });
+
+  const markers = state.markers.map((marker, index) => {
+    const opacityStyle = {
+      opacity: interpolations[index].opacity,
+    };
+    return (
+      <Marker key={marker.id} coordinate={marker.coordinate}>
+        <Animated.View style={[styles.markerWrap, opacityStyle]}>
+          <View style={styles.marker} />
+        </Animated.View>
+      </Marker>
+    );
+  });
+
   return (
     <View style={styles.container}>
-      <Text>Map Screen</Text>
-      <Button title="Test" onPress={test} />
+      <MapView
+        ref={mapRef}
+        initialRegion={state.region}
+        style={styles.container}
+      >
+        {markers}
+      </MapView>
+      <Animated.ScrollView
+        horizontal
+        scrollEventThrottle={1}
+        showsHorizontalScrollIndicator={false}
+        snapToInterval={CARD_WIDTH}
+        onScroll={Animated.event(
+          [
+            {
+              nativeEvent: {
+                contentOffset: {
+                  x: mapRef.animation,
+                },
+              },
+            },
+          ],
+          { useNativeDriver: true },
+        )}
+        style={styles.scrollView}
+        contentContainerStyle={styles.endPadding}
+      >
+        {state.markers.map((marker) => (
+          // TODO: change this temporary id
+          <View style={styles.card} key={marker.id}>
+            <Image
+              source={marker.image}
+              style={styles.cardImage}
+              resizeMode="cover"
+            />
+            <View style={styles.textContent}>
+              <Text numberOfLines={1} style={styles.cardTitle}>
+                {marker.title}
+              </Text>
+              <Text numberOfLines={1} style={styles.cardDescription}>
+                {marker.description}
+              </Text>
+            </View>
```

Comment:
Same here, separating out to a separate component file would be nicer for modularity